nabs/event.py

```diff
@@ -0,0 +1,109 @@
+from collections import defaultdict
+from functools import partial
+from threading import Event, Thread
+import math
+import time
+
+from ophyd.status import Status
+
+
+class EventBuilder:
+    def __init__(self, signals, rate, timeout=10.0, duplicates='closest'):
```

Comment:
`Duplicates` should be an Enum? Also maybe a class attribute? 🤷‍♂️

Reply:
Probably correct, if only so that the choices available are clear.